packages/@uppy/companion-client/src/RequestClient.js

```diff
@@ -183,18 +197,35 @@ export default class RequestClient {
   /** @protected */
   async request ({ path, method = 'GET', data, skipPostResponse, signal }) {
     try {
-      const headers = await this.preflightAndHeaders(path)
-      const response = await fetchWithNetworkError(this.#getUrl(path), {
-        method,
-        signal,
-        headers,
-        credentials: this.opts.companionCookiesRule || 'same-origin',
-        body: data ? JSON.stringify(data) : null,
-      })
-      if (!skipPostResponse) this.onReceiveResponse(response)
-      return handleJSONResponse(response)
+      return await pRetry(async () => {
```

Comment:
Just a head up that is will be need to be refactored again in #4745, as we now have a util for fetching and retrying.